platform/ui/src/utils/ModalProvider.js

```diff
@@ -0,0 +1,91 @@
+import React, { useState, createContext, useContext } from 'react';
+import Modal from 'react-bootstrap-modal';
+
+const ModalContext = createContext(null);
+const { Provider, Consumer } = ModalContext;
+
+export const useModalContext = () => useContext(ModalContext);
+
+const ModalProvider = ({ children }) => {
+  const DEFAULT_OPTIONS = {
+    component: null,
+    backdrop: false,
+    keyboard: false,
+    show: true,
+    large: true,
+    closeButton: true,
+    title: null,
+    customClassName: null,
+  };
+
+  const [options, setOptions] = useState(DEFAULT_OPTIONS);
+
+  const show = (component, props = {}) =>
+    setOptions({
+      ...options,
+      component,
+      ...props,
+    });
+
+  const hide = () => setOptions(DEFAULT_OPTIONS);
+
+  return (
+    <Provider value={{ ...options, show, hide }}>
+      <Consumer>
+        {props => {
+          const {
+            component: Component,
+            footer: Footer,
+            header: Header,
+            body: Body,
+            customClassName,
+          } = props;
+          return Component || Body ? (
+            <Modal
+              className={`modal fade themed in ${customClassName} ${
+                Component ? Component.className : Body.className
+              }`}
+              backdrop={options.backdrop}
+              keyboard={options.keyboard}
+              show={options.show}
+              large={options.large}
+              title={options.title}
+              onHide={hide}
+            >
+              {(Header || options.title) && (
+                <Modal.Header closeButton={options.closeButton}>
+                  {options.title && <Modal.Title>{options.title}</Modal.Title>}
+                  {Header && <Header hide={hide} />}
+                </Modal.Header>
+              )}
+              <Modal.Body>
+                {Body && <Body hide={hide} />}
+                {Component && <Component hide={hide} />}
+              </Modal.Body>
+              {Footer && (
+                <Modal.Footer>
+                  {' '}
+                  <Footer hide={hide} />
+                </Modal.Footer>
+              )}
+            </Modal>
+          ) : null;
+        }}
+      </Consumer>
+      {children}
```

Comment:
So `children` would render outside of the modal?

Reply:
Yes, this children is the rest of the app that goes inside the provider itself.

Reply:
Ahh! That makes a lot of sense.
Instead of baking the Modal into the provider, can we keep it as a separate `@ohif/ui` modal that we place beside the `<OHIFStandaloneViewer>` in `@ohif/viewer`?

https://github.com/OHIF/Viewers/pull/1110/files#diff-d14361ccec1e91c0c2716e30087fe0f2R106

This allows for anyone reusing our `@ohif/ui` but with a custom viewer project to swap out the implementation with anything that's API compatible.

Reply:
I'm passing it as a prop to the provider now. Thx!